frontend/src/services/user_groups.ts

```diff
@@ -0,0 +1,45 @@
+import { ListObjectForAdminQuery, PaginationResult, UserGroup, UserGroupAdminView } from 'src/global_types'
+import { backendDataSource as ds } from './data_sources/backend'
+
+export async function listUserGroups(i: {
+  query: string,
+  includeDeleted?: boolean,
+  operationSlug: string
+}): Promise<Array<UserGroup>> {
+  return await ds.listUserGroups(i.query, i.includeDeleted || false, i.operationSlug)
+}
+
+export async function listUserGroupsAdminView(i: ListObjectForAdminQuery): Promise<PaginationResult<UserGroupAdminView>> {
+  return await ds.adminListUserGroups(i)
+}
+
+export async function createUserGroup(i: {
+  name: string,
+  userSlugs: string[],
+}): Promise<void> {
+  let slug = i.name.toLowerCase().replace(/[^A-Za-z0-9]+/g, '-').replace(/^-|-$/g, '')
```

Comment:
Suppose my user group name was `$$bestgroup$$`, I think that would initially change this to `--bestgroup--`, then get fixed into just `-bestgroup-`? 
If that's the case, then...
It looks like we're trying to prevent user groups from starting/trailing -'s, so maybe we want to change the latter regex to `/^-+|-+$/g` (or collapse all multiple -'s into a single dash before doing the final stripping)

Reply:
that's a good point! This function is actually some copypasta from `createOperation` in `operations.ts` but slightly modified for usergroups. yeah I'm down for changing it (I guess in both places) to `/^-+|-+$/g`

Reply:
changed